healthcheck/healthcheck.go

```diff
@@ -46,12 +49,20 @@ type loadBalancer interface {
 }
 
 func newHealthCheck() *HealthCheck {
-	return &HealthCheck{make(map[string]*BackendHealthCheck), nil}
+	return &HealthCheck{
+		Backends: make(map[string]*BackendHealthCheck),
+		wg:       sync.WaitGroup{},
+	}
 }
 
 // NewBackendHealthCheck Instantiate a new BackendHealthCheck
 func NewBackendHealthCheck(URL string, interval time.Duration, lb loadBalancer) *BackendHealthCheck {
-	return &BackendHealthCheck{URL, interval, nil, lb}
+	return &BackendHealthCheck{
+		URL:            URL,
+		Interval:       interval,
+		requestTimeout: 5 * time.Second,
```

Comment:
That's Go's way of saying "5 seconds".

(Technically, durations are just float type aliases.)

5 secs was the default before, so I kept it but made it configurable for testing purposes.